### ISSUE TITLE
How to install Jenkins to build and test Dataverse

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse-jenkins#1]
Last week @donsizemore set up https://jenkins.dataverse.org (screenshot below) for the Dataverse community but I'd like to know how to install Jenkins myself and start building and testing Dataverse. I imagine that anyone who runs a fork of Dataverse would want this as well. Knowing Don, he's using Ansible to install and configure Jenkins. It would be great to have the playbook and a README to explain how to use it. Thanks in advance!

<img width="1096" alt="Screen Shot 2019-04-08 at 9 27 51 AM" src="https://user-images.githubusercontent.com/21006/55727754-a65a6880-59e0-11e9-8080-c29694e2a26c.png">

**Comment by pdurbin** [comment]
@donsizemore another thought is to add a Vagrantfile as well as a README. This is what I call "executable documentation": 😄 

<img width="755" alt="Screen Shot 2019-04-08 at 10 09 49 AM" src="https://user-images.githubusercontent.com/21006/55730577-86c63e80-59e6-11e9-8254-ed0dde2b3f46.png">

From https://www.slideshare.net/philipdurbin/how-to-run-a-successful-open-source-java-ee-project via http://wiki.greptilian.com/talks/2017/javaone-how-to-run-an-open-source-project/

**Comment by pdurbin** [comment]
http://itzikb.com/installing-jenkins-using-ansible/ looks helpful. I just got "Welcome to Jenkins" at http://ec2-35-153-208-116.compute-1.amazonaws.com:8080 like this...

![Screen Shot 2019-06-04 at 4 32 56 PM](https://user-images.githubusercontent.com/21006/58911730-75e43180-86e6-11e9-894b-13a915265fd7.png)

... by doing this:

```
[root@ip-172-31-39-101 ~]# vim jenkins.yml
[root@ip-172-31-39-101 ~]# cat jenkins.yml
- name: Installing Jenkins
  hosts: localhost
  roles:
     - { role: geerlingguy.jenkins, jenkins_plugins: "" }
[root@ip-172-31-39-101 ~]# 
```

[jenkins.yml.txt](https://github.com/IQSS/dataverse-jenkins/files/3258484/jenkins.yml.txt)

```
yum install /usr/bin/ansible-playbook # ansible-2.4.2.0-2.el7.noarch
ansible-galaxy install -p roles geerlingguy.jenkins
ansible-playbook jenkins.yml
```

I can log in with admin/admin but when I try to create a new job...

![Screen Shot 2019-06-04 at 4 36 51 PM](https://user-images.githubusercontent.com/21006/58911954-0589e000-86e7-11e9-970c-6a2749dd6812.png)

... I'm not sure how to get `config.xml` from this repo into it.

**Comment by pdurbin** [comment]
@donsizemore if you could be so kind, please take a look at pull request #6 which I just opened. Thanks!

**Comment by pdurbin** [comment]
Here's the script I using to spin up EC2 instances to install Jenkins on:

[ec2-create-instance.sh.txt](https://github.com/IQSS/dataverse-jenkins/files/3266229/ec2-create-instance.sh.txt)

This is almost the same script as the one at http://guides.dataverse.org/en/4.14/developers/deployment.html#download-and-run-the-create-instance-script

I'm planning on cleaning it up and adding it to this repo.

**Comment by donsizemore** [comment]
do #15 and #16 address this?

**Comment by pdurbin** [comment]
@donsizemore probably. Maybe I'll use this issue to represent one final test of all the stuff we just merged in pull request #16. 😄

**Comment by pdurbin** [comment]
@donsizemore meh. At standup today we decided to close this issue without extra testing. Nothing has changed on master since I tested in my branch. Thanks so much for making it so easy to spin up Jenkins to test Dataverse and friends!
